SOS.py

```diff
@@ -52,21 +52,17 @@ def getSOSByHourSingle():
     
     # Get data for restraunt N days ago
     # Lock for potential error with updater thread
-    newDataLock.acquire()
-    day = getSinceDate(
-        datetime.strftime((datetime.now() - timedelta(n)), "%Y-%m-%d"), restNmbr
-    )
-    newDataLock.release()
-    
-    # If data hasn't been fetched yet, try to update.  
-    if len(day) == 0:
-        update()
-        newDataLock.acquire()
+    with newDataLock:
         day = getSinceDate(
             datetime.strftime((datetime.now() - timedelta(n)), "%Y-%m-%d"), restNmbr
         )
-        newDataLock.release()
     
+        # If data hasn't been fetched yet, try to update.
+        if len(day) == 0:
+            update()
+            day = getSinceDate(
+                datetime.strftime((datetime.now() - timedelta(n)), "%Y-%m-%d"), restNmbr
+            )
```

Comment:
The lock is no longer released before checking whether we need to fetch data. This is to avoid a race condition where we fetch data twice, once from here and once from the updater thread.